pkg/ruler/rule_query_consistency.go

```diff
@@ -4,28 +4,37 @@ package ruler
 
 import (
 	"context"
+	"fmt"
 	"time"
 
+	"github.com/go-kit/log"
 	"github.com/prometheus/prometheus/model/labels"
 	"github.com/prometheus/prometheus/promql"
 	"github.com/prometheus/prometheus/rules"
 	"github.com/prometheus/prometheus/storage"
 	"github.com/prometheus/prometheus/util/annotations"
 
 	"github.com/grafana/mimir/pkg/querier/api"
+	"github.com/grafana/mimir/pkg/util/spanlogger"
 )
 
+const alertForStateMetricName = "ALERTS_FOR_STATE"
+
 // WrapQueryFuncWithReadConsistency wraps rules.QueryFunc with a function that injects strong read consistency
 // requirement in the context if the query is originated from a rule which depends on other rules in the same
 // rule group.
-func WrapQueryFuncWithReadConsistency(fn rules.QueryFunc) rules.QueryFunc {
+func WrapQueryFuncWithReadConsistency(fn rules.QueryFunc, logger log.Logger) rules.QueryFunc {
 	return func(ctx context.Context, qs string, t time.Time) (promql.Vector, error) {
+
 		// Get details about the rule.
 		detail := rules.FromOriginContext(ctx)
 
 		// If the rule has dependencies then we should enforce strong read consistency,
 		// otherwise we'll fallback to the per-tenant default.
 		if !detail.NoDependencyRules {
+			spanLog := spanlogger.FromContext(ctx, logger)
+			spanLog.DebugLog("msg", "forced strong read consistency because the rule depends on other rules in the same rule group")
```

Comment:
Would adding a tag / attribute to the span help so that we could query by it?

Reply:
Is https://github.com/grafana/mimir/pull/7193/commits/e3525fb5864a0be93ba94c7a04d6f18b0ccb75d0 what you had in mind?

Reply:
Exactly, thanks!